ngraph/core/include/ngraph/descriptor/tensor.hpp

```diff
@@ -49,10 +54,19 @@ namespace ngraph
             void set_tensor_type(const element::Type& element_type, const PartialShape& pshape);
             void set_element_type(const element::Type& elemenet_type);
             void set_partial_shape(const PartialShape& partial_shape);
+            void set_lower_value(const HostTensorPtr& value);
+            void set_upper_value(const HostTensorPtr& value);
+            void invalidate_values();
 
             const element::Type& get_element_type() const { return m_element_type; }
             const Shape& get_shape() const;
             const PartialShape& get_partial_shape() const { return m_partial_shape; }
+            HostTensorPtr get_lower_value() const { return m_lower_value; }
+            HostTensorPtr get_upper_value() const { return m_upper_value; }
+            bool has_and_set_bound() const
+            {
+                return m_upper_value != nullptr && m_upper_value == m_lower_value;
```

Comment:
This is expected. It is cheap comparison of values.

Reply:
@GlebKazantaev and @ilyachur 
We designed methos like that intentionally. We need to check if bounds are the same too often, so the process of validation, if actual values are equal would take too much time. 
So the whole value propagation process is designed to store the same shared_ptr in those m_upper_value and m_lower_value fields to make checking nearly costless.

Reply:
Discussed offline that this logic is expected (comments and documentation will be added) otherwise HostTensor comparison will significantly impact LoadTime.